In [4]:
import csv
f = open("guns.csv", "r")
reader = csv.reader(f)
data = list(reader)
print(data[:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [5]:
headers = data[0]
data = data[1:]
print(headers)
print(data[:5])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [6]:
years = []
for row in data:
    years.append(row[1])
    
year_counts = {}
for year in years:
    if year not in year_counts:
        year_counts[year] = 1
    else:
        year_counts[year] += 1

year_counts

{'2012': 33563, '2013': 33636, '2014': 33599}

In [7]:
import datetime
dates = []
for row in data:
    year = int(row[1])
    month = int(row[2])
    date = datetime.datetime(year=year, month=month, day=1)
    dates.append(date)
dates[:5]

[datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0)]

In [8]:
date_counts = {}
for date in dates:
    if date not in date_counts:
        date_counts[date] = 1
    else:
        date_counts[date] += 1
date_counts

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [9]:
sex_counts = {}
for row in data:
    sex = row[5]
    if sex not in sex_counts:
        sex_counts[sex] = 1
    else:
        sex_counts[sex] += 1
sex_counts

{'F': 14449, 'M': 86349}

In [10]:
race_counts = {}
for row in data:
    race = row[7]
    if race not in race_counts:
        race_counts[race] = 1
    else:
        race_counts[race] += 1
race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

*What we have learned so far*
In this dataset, far more men (86,349) than women (14,449) are killed by guns.  The largest racial group in the gun deaths dataset is White (66,237), followed by Black (23,296).  Hispanic is a distant third, at 9,022 deaths.  This may in part be a result of the overall ethnic makeup of the United States.

The month with the lowest gun violence seems to be February.  Across years, there is no visible major difference.

Further avenues for exploration include:
- Deaths by intent
- Deaths by educational background
- Deaths by location

Each of the above could also be sorted by sex, age, and race.

In [12]:
f = open("census.csv", "r")
reader2 = csv.reader(f)
census = list(reader2)

In [13]:
census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

In [23]:
mapping = {
    "Asian/Pacific Islander": int(census[1][14]) + int(census[1][15]),
    "Black": census[1][12],
    "Native American/Native Alaskan": census[1][13],
    "Hispanic": census[1][11],
    "White": census[1][10]
}
mapping
    

{'Asian/Pacific Islander': 15834141,
 'Black': '40250635',
 'Hispanic': '44618105',
 'Native American/Native Alaskan': '3739506',
 'White': '197318956'}

In [42]:
race_per_hundredk = {}
for key, value in (race_counts.items()):    
    race_per_hundredk[key] = int(value)*100000/int(mapping[key])
race_per_hundredk
                            

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.877347773519595,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

In [47]:
intents = []
for row in data:
    intents.append(row[3])
intents

['Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Undetermined',
 'Suicide',
 'Accidental',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Homicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Undetermined',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homicide',
 'Homicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Undetermined',
 'Suicide',
 'Suicide',
 'Accidental',
 'Homicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Homicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Suicide',
 'Accidental',
 'Suicide',
 'Accidental',
 'Suicide',
 'Undetermined',
 'Suicide',
 'Suicide',
 'Suicide',
 'Homi

In [48]:
races = []
for row in data:
    races.append(row[7])
races

['Asian/Pacific Islander',
 'White',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Black',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'Asian/Pacific Islander',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Black',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'White',
 'Native American/Native Alaskan',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'Native American/Native Alaskan',
 'White',
 'White',
 'Native American/Native Alaskan',
 'Asian/Pacific Islander',
 'White',
 'Native American/Native Alaskan',
 'Native American/Native Alaskan',
 'White',
 'White',
 'White',
 'Native American/Native Alaskan',
 'Asian/Pacific Islander',
 'White',
 'White',
 'Native Ame

In [49]:
homicide_race_counts = {}
for i, race in enumerate(races):
    if intents[i] == 'Homicide':
        if race not in homicide_race_counts:
            homicide_race_counts[race] = 0
        homicide_race_counts[race] += 1
homicide_race_counts

{'Asian/Pacific Islander': 559,
 'Black': 19510,
 'Hispanic': 5634,
 'Native American/Native Alaskan': 326,
 'White': 9147}

In [50]:
for key, value in (race_counts.items()):    
    homicide_race_counts[key] = int(value)*100000/int(mapping[key])
homicide_race_counts

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.877347773519595,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

# Findings

Rankings of homicide rates by race:
1) Black (58 per 100,000)
2) White (34 per 100,000)
3) Native American/Naive Alaskan (25 per 100,000)
4) Hispanic (20 per 100,000)
5) Asian/Pacific Islander (8 per 100,000)

Next Steps:
Canadian data on homicide by race!